Python/codecs.c

```diff
@@ -49,36 +49,36 @@ int PyCodec_Register(PyObject *search_function)
     return -1;
 }
 
+extern int _Py_normalize_encoding(const char *, char *, size_t);
+
 /* Convert a string to a normalized Python string: all characters are
-   converted to lower case, spaces are replaced with underscores. */
+   converted to lower case, spaces and hyphens are replaced with underscores. */
 
 static
 PyObject *normalizestring(const char *string)
 {
-    size_t i;
     size_t len = strlen(string);
-    char *p;
+    char *encoding;
     PyObject *v;
 
     if (len > PY_SSIZE_T_MAX) {
         PyErr_SetString(PyExc_OverflowError, "string is too large");
         return NULL;
     }
 
-    p = PyMem_Malloc(len + 1);
-    if (p == NULL)
+    encoding = PyMem_Malloc(len + 1);
+    if (encoding == NULL)
         return PyErr_NoMemory();
-    for (i = 0; i < len; i++) {
-        char ch = string[i];
-        if (ch == ' ')
-            ch = '-';
-        else
-            ch = Py_TOLOWER(Py_CHARMASK(ch));
-        p[i] = ch;
+
+    if (!_Py_normalize_encoding(string, encoding, len + 1))
+    {
+        Py_FatalError("normalizestring(): encoding failed");
```

Comment:
Or use a regular exception:

```suggestion
        PyErr_SetString(PyExc_RuntimeError, "_Py_normalize_encoding() failed");
```